AspNetCore3ChildlifetimeScope/SampleControllerApplicationB.cs

```diff
@@ -0,0 +1,35 @@
+using AspNetCore3ChildlifetimeScope.Dependencies;
+using Autofac;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Hosting;
+
+namespace AspNetCore3ChildlifetimeScope
+{
+    [ApiController]
+    [Route("ApplicationB")]
+    public class SampleControllerApplicationB : ControllerBase
+    {
+        private readonly IApplicationBScopedDependency applicationBScopedDependency;
+        private readonly ITransientRootChildScopeDependency transientRootChildScopeDependency;
+        private readonly ISingletonRootDependency singletonRootDependency;
+
+        public SampleControllerApplicationB(IApplicationBScopedDependency applicationBScopedDependency,
+            ITransientRootChildScopeDependency transientRootChildScopeDependency,
+            ISingletonRootDependency singletonRootDependency)
+        {
+            this.applicationBScopedDependency = applicationBScopedDependency;
+            this.transientRootChildScopeDependency = transientRootChildScopeDependency;
+            this.singletonRootDependency = singletonRootDependency;
+        }
+
+        [HttpGet]
+        public IActionResult ShowMessage() => this.Ok(new
+        {
+            Message =
+                $"Resolved {nameof(IApplicationBScopedDependency)} from application-container. Resolved {nameof(ITransientRootChildScopeDependency)} from {nameof(ILifetimeScope)} passed to the {nameof(IHost)}. Resolved {nameof(ISingletonRootDependency)} from the container itself.",
+            ApplicationBScopedDependency = this.applicationBScopedDependency.GetHashCode(),
```

Comment:
It might be better to write out the actual type names, like `this.applicationBScopedDependency.GetType()`, rather than using `nameof` here. The `GetHashCode` method isn't going to show anyone anything useful.

Perhaps the objects themselves need some sort of way to identify, like a name or some metadata to tie things back and have them be able to self-identify as being "from this specific registration."